doc/Learn/Tutorials/Navigation/Advanced/HowTo-UseTabBar.md

```diff
@@ -164,20 +161,186 @@ The Navigation capabilities offered by Uno.Extensions include regions. Regions a
+      xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+      xmlns:local="using:Uno.Extensions.Navigation.UI.Samples"
+      xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+      xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+      mc:Ignorable="d"
+      Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+  
+      <Grid>
+            <StackPanel Orientation="Vertical">
+                <TextBlock Text="Benefits of subscribing:"
+                            FontSize="24"
+                            HorizontalAlignment="Center"
+                            VerticalAlignment="Center" />
+                <Button Content="Sign Up"
+                        HorizontalAlignment="Center"
+                        VerticalAlignment="Center" />
+            </StackPanel>
+      </Grid>
+  </Page>
+  ```
+
+* For the purposes for this tutorial, `SignUpPage` will be associated with its own view model `SignUpViewModel`. Add a new **Class** item to your app called `SignUpViewModel` with the following code:
+
+  ```csharp 
+  namespace UsingTabBar.ViewModels;
+  
+  public class SignUpViewModel
+  {
+      public SignUpViewMode()
+      {
+
+      }
+  }
+  ```
+
+* Register `ViewMap` and `RouteMap` instances inside the `RegisterRoutes` method in `AppHead.xaml.cs`. This associates the `SignUpPage` described above with `SignUpViewModel`, as well as avoiding the use of reflection for route discovery.
+
+    ```csharp
+    private static void RegisterRoutes(IViewRegistry views, IRouteRegistry routes)
+    {
+        views.Register(
+            new ViewMap<ShellControl, ShellViewModel>(),
+            new ViewMap<LoginPage, LoginViewModel>(),
+            new ViewMap<MainPage, MainViewModel>()
+        );
+    
+        routes.Register(
+            new RouteMap("", View: views.FindByViewModel<ShellViewModel>(),
+                Nested: new RouteMap[]
+                {
+                    new RouteMap("Main", View: views.FindByViewModel<MainViewModel>(),
+                    Nested: new RouteMap[]
+                    { 
+                        new RouteMap("Login", View: views.FindByViewModel<LoginViewModel>())
+                    })
+                }));
+    }
+    ```
+  
+* Importantly, the snippet above establishes a route name `SignUp` for `SignUpPage`. We can use this route name to navigate to the `SignUpPage` view element.
+
+* Add a `TabBarItem` to the `TabBar` element with the `uen:Region.Name` attached property set to `SignUp`.
+
+    ```xml
+    <utu:TabBar.Items>
+        <utu:TabBarItem uen:Region.Name="One" 
+                        Style="{StaticResource MaterialBottomTabBarItemStyle}" />
+        <utu:TabBarItem uen:Region.Name="Two" 
+                        Style="{StaticResource MaterialBottomTabBarItemStyle}" />
+        <utu:TabBarItem uen:Region.Name="Three" 
+                        Style="{StaticResource MaterialBottomTabBarItemStyle}" />
```

Comment:
@Youssef1313 These represent regions defined in XAML which do not require routes to be registered

Reply:
Well, if I understand correctly, `uen:Region.Name="Something"` can be used for two things:

- It can be defined in a navigation element, e.g, TabBarItem, to define a route that will be navigated to.
- It can be defined in other elements to say "when asked to navigate to "Something", show this part of the UI.

I think clear rules as to when it means which is needed. Also there is a possibility for conflicts where the region name to be navigated to is defined both in another element in XAML and also in routes. If there are any precedence rules, it might be worth documenting.

Reply:
In most cases if you want to trigger navigation you would use Navigation.Request to define the route you want to navigate to.
For selector controls (TabBar and NavigationView) where you have selectable items you annotate them with Region.Name - this is used to determine what item to select. For example if you have a tabbaritem with Region.Name="Products" when navigating with Request.Route = Products, navigation knows to select that tab item
Furthermore, in the content area (Grid with Region.Navigator="Visibility") navigation will first look for children with Region.Name="Products" and failing that, it'll look up the route map table to determine if there's a page or control for the Products route. If the view is a page, navigation will add a frame to the grid and then navigate to the page; if the view is any other control it's simply created and added to the grid. In both cases, the element being added to the grid is annotated with Region.Name="Products" to prevent a new element being created every time the tab item is selected.